### PR TITLE
2D Visibility, new package, GSoC 2013

### PR DESCRIPTION
**Pull request by mhsaar**
New Feature:
https://cgal.geometryfactory.com/CGAL/Members/wiki/Features/Visibility_2

tested on:
https://cgal.geometryfactory.com/CGAL/Members/testsuite/results-4.7-Ic-61.shtml

### CODE REVIEW

**Comment by lrineau** [comment]
Can you please refresh the branch and fix the conflicts with `master`?

**Comment by lrineau** [comment]
* ;x; There is a directory `Visibility_2/data`. What is its purpose? It must be removed from the branch, and probably even all its history, if that makes sense.
* :x: The compressed size of the branch is big: 2.1MB. That is probably due to that spurious and unwanted directory `data/`.
* :white_check_mark: I agree the results in [CGAL-4.7-Ic-61](https://cgal.geometryfactory.com/CGAL/Members/testsuite/results-4.7-Ic-61.shtml#Visibility_2) are really good.

**Comment by mhsaar** [comment]
The data was for benchmarks. How can I remove that from history ?

I'll check for conflicts afterwards.

On Tue, Jun 30, 2015 at 11:19 AM, Laurent Rineau <notifications@github.com>
wrote:

>
>    - ;x; There is a directory Visibility_2/data. What is its purpose? It
>    must be removed from the branch, and probably even all its history, if that
>    makes sense.
>    - [image: :x:] The compressed size of the branch is big: 2.1MB. That
>    is probably due to that spurious and unwanted directory data/.
>    - [image: :white_check_mark:] I agree the results in CGAL-4.7-Ic-61
>    <https://cgal.geometryfactory.com/CGAL/Members/testsuite/results-4.7-Ic-61.shtml#Visibility_2>
>    are really good.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CGAL/cgal/pull/145#issuecomment-117070325>.
>



-- 

Dr. Michael Hemmer

Phone: +49 531-391-3116
Fax:     +49 531/391-3109
Email:  mhsaar@gmail.com
Web:    http://www.ibr.cs.tu-bs.de/users/hemmer/
--------------------------------------------------------------------

**Comment by lrineau** [comment]
I think the command would be:

    git filter-branch --tree-filter 'rm -rf Visibility_2/data/' master..HEAD

Explanation:
  - `--tree-filter` describes a shell command to run on each commit, before re-writing it,
  - `master..HEAD` is the list of commits that are in the current branch (designed by the symbolic reference `HEAD`, and are not in `master`.

@sloriot, can you confirm?

**Comment by mhsaar** [comment]
done

but it seems that I don't have to commit/push this anymore

On Tue, Jun 30, 2015 at 11:41 AM, Laurent Rineau <notifications@github.com>
wrote:

> I think the command would be:
>
> git filter-branch --tree-filter 'rm -rf Visibility_2/data/' master..HEAD
>
> Explanation:
>
>    - --tree-filter describes a shell command to run on each commit,
>    before re-writing it,
>    - master..HEAD is the list of commits that are in the current branch
>    (designed by the symbolic reference HEAD, and are not in master.
>
> @sloriot <https://github.com/sloriot>, can you confirm?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CGAL/cgal/pull/145#issuecomment-117095390>.
>



-- 

Dr. Michael Hemmer

Phone: +49 531-391-3116
Fax:     +49 531/391-3109
Email:  mhsaar@gmail.com
Web:    http://www.ibr.cs.tu-bs.de/users/hemmer/
--------------------------------------------------------------------

**Comment by lrineau** [comment]
That is strange. It should have created new commits for all commits of your branch. That should have created a deviation between your local copy of `gsoc2013-Visibility_doc-hemmer` and the ones in the remotes repositories.

Sébastien knows more than me about `git-filter-branch`, but he is on vacation today and tomorrow.

**Comment by mhsaar** [comment]
Yes there was a deviation.
But I was not able to commit it.
Instead "git status" said that there is a deviation and I should do a pull.
I got back the data files.
I can remove them again, but not sure how to get that on to server..



On Tue, Jun 30, 2015 at 4:39 PM, Laurent Rineau <notifications@github.com>
wrote:

> That is strange. It should have created new commits for all commits of
> your branch. That should have created a deviation between your local copy
> of gsoc2013-Visibility_doc-hemmer and the ones in the remotes
> repositories.
>
> Sébastien knows more than me about git-filter-branch, but he is on
> vacation today and tomorrow.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CGAL/cgal/pull/145#issuecomment-117211055>.
>



-- 

Dr. Michael Hemmer

Phone: +49 531-391-3116
Fax:     +49 531/391-3109
Email:  mhsaar@gmail.com
Web:    http://www.ibr.cs.tu-bs.de/users/hemmer/
--------------------------------------------------------------------

**Comment by lrineau** [comment]
Hi Michael, I will handle that. I have added you to the collaborators of my CGAL fork http://github.com/lrineau/cgal (that is *my* `mine` repository). You will push your version of the branch on your machine in my fork, and I will sort-out the issues.
```
git remote add lrineau git@github.com:lrineau/cgal.git
git fetch lrineau
git push lrineau gsoc2013-Visibility_doc-hemmer
```

Please also add me (@lrineau) as a collaborator in your fork https://github.com/mhsaar/cgal ; the setting page for that is https://github.com/mhsaar/cgal/settings/collaboration

**Comment by sloriot** [comment]
You were probably missing a -f to force the push.

**Comment by mhsaar** [comment]
ok, where ?
can you give the complete line ?
but it would also be ok if you do it?

I am back to my computer at 12 the earliest.

On Thu, Jul 2, 2015 at 7:20 AM, Sebastien Loriot <notifications@github.com>
wrote:

> You were probably missing a -f to force the push.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CGAL/cgal/pull/145#issuecomment-117909404>.
>



-- 

Dr. Michael Hemmer

Phone: +49 531-391-3116
Fax:     +49 531/391-3109
Email:  mhsaar@gmail.com
Web:    http://www.ibr.cs.tu-bs.de/users/hemmer/
--------------------------------------------------------------------

**Comment by mhsaar** [comment]
done



On Wed, Jul 1, 2015 at 2:54 PM, Laurent Rineau <notifications@github.com>
wrote:

> Hi Michael, I will handle that. I have added you to the collaborators of
> my CGAL fork http://github.com/lrineau/cgal
> <https://github.com/lrineau/cgal> (that is *my* mine repository). You
> will push your version of the branch on your machine in my fork, and I will
> sort-out the issues.
>
> git remote add lrineau git@github.com:lrineau/cgal.git
> git fetch lrineau
> git push lrineau gsoc2013-Visibility_doc-hemmer
>
> Please also add me (@lrineau <https://github.com/lrineau>) as a
> collaborator in your fork https://github.com/mhsaar/cgal ; the setting
> page for that is https://github.com/mhsaar/cgal/settings/collaboration
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/CGAL/cgal/pull/145#issuecomment-117650877>.
>



-- 

Dr. Michael Hemmer

Phone: +49 531-391-3116
Fax:     +49 531/391-3109
Email:  mhsaar@gmail.com
Web:    http://www.ibr.cs.tu-bs.de/users/hemmer/
--------------------------------------------------------------------

**Comment by lrineau** [comment]
With the help of Sébastien, the branch in `mhsaar/` is now modified: filtered with `git-filter-branch`, like what you did, Michael.

The weight of the branch is now acceptable.

Michael, now, on your side, in a checkout of your local copy of `gsoc2013-Visibility_doc-hemmer`, you must do:
```
git fetch mine
git reset --hard mine/gsoc2013-Visibility_doc-hemmer
```
(similar to what we do with `cgal-dev/integration`).


There are still conflicts between the branch and `master`, in those files:
```
        both modified:   Documentation/biblio/cgal_manual.bib
        both modified:   Documentation/biblio/geom.bib
        both modified:   Installation/changes.html
```

I do not know what is the best solution. Anyway, a lot of pull-requests do modify those files, by adding things at the end (instead of respecting the alphabetical order). That explains why there are so many conflicts.

**Comment by afabri** [comment]
Hello,
As it cannot merged automatically, someone must solve the merge conflicts: you :>
In order not to repeat this at every merge into integration, recreate your branch from master as described [here] (https://cgal.geometryfactory.com/CGAL/Members/wiki/Guidelines/Source_Code_Management_GIT/FAQ#My_branch_is_really_old._I_would_like_to_update_it_to_get_latest_update_from_master.).

**Comment by sloriot** [comment]
Merged in [CGAL-4.7-Ic-80](https://cgal.geometryfactory.com/CGAL/Members/testsuite/results-4.7-Ic-80.shtml)

**Comment by sloriot** [comment]
Test results look good.

File: Visibility_2/doc/Visibility_2/PackageDescription.txt [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+\cgalPkgDescriptionBegin{2D Visibility Computation,PkgVisibility_2Summary}
+\cgalPkgPicture{visibility-teaser-thumbnail.png}
+\cgalPkgSummaryBegin
+\cgalPkgAuthors{Michael Hemmer, Kan Huang, Francisc Bungiu, Ning Xu}
+\cgalPkgDesc{This package provides several variants to compute 
+the visibility area of a point within polygonal regions in two dimensions.}
+\cgalPkgManuals{Chapter_2D_Visibility_Computation,PkgVisibility_2}
+\cgalPkgSummaryEnd
+\cgalPkgShortInfoBegin
+\cgalPkgSince{99.99}
```
Comment (lrineau) [inline Visibility_2/doc/Visibility_2/PackageDescription.txt]: @sloriot: is that the "Introduced in" you were talking about?

Comment (sloriot) [inline Visibility_2/doc/Visibility_2/PackageDescription.txt]: yep

**Comment by lrineau** [comment]
Congratulation, Michael (@mhsaar). Your package will be in CGAL-4.7!

... and then you leave. :-\

**Comment by mhsaar** [comment]
Great, thanks.

**Pull request merged**